### ISSUE TITLE
Update signals' value in `onMount` doesn't re-render nested routes

### ISSUE THREAD

**Issue by yume-chan** [solidjs/solid-start#917]
1. In a route layout, created a resource.
2. Pass the resource down using a context.
3. Read `resource()` or `resource.latest` in a nested route.
4. Call `mutate` in layout's `onMount`
5. The rendered value in nested route doesn't change

Repro:

https://github.com/yume-chan/solid-start-issue

```tsx
// src/routes/a.tsx

const MyContext = createContext<InitializedResource<string>>();

export function useMyContext() {
  const value = useContext(MyContext);
  if (!value) {
    throw new Error("MyContext not found");
  }
  return value;
}

function Child(props: { value: InitializedResource<string> }) {
  return <div>Child: {props.value.latest}</div>;
}

export default function Layout() {
  const [value, { mutate }] = createResource(
    () => { },
    () => "fetcher",
    { initialValue: "initial" }
  );

  onMount(() => {
    mutate("mutate");
  });

  return (
    <MyContext.Provider value={value}>
      <div>Layout: {value.latest}</div>
      <Child value={value} />
      <Outlet />
    </MyContext.Provider>
  );
}
```

```tsx
// src/routes/a/index.tsx

export default function Page() {
  const value = useMyContext();
  return <div>Page: {value.latest}</div>;
}
```

Result:

![image](https://github.com/solidjs/solid-start/assets/1330321/e85a68dc-467c-431e-bb6c-64758e92ad54)

**Comment by yume-chan** [comment]
In fact, it doesn't need to be a resource, calling setter of `createSignal` also doesn't re-render nested routes.

See the https://github.com/yume-chan/solid-start-issue/tree/signal branch.

```ts
// src/routes/a.tsx
import {
  Accessor,
  createContext,
  createSignal,
  onMount,
  useContext
} from "solid-js";
import { Outlet } from "solid-start";

const MyContext = createContext<Accessor<string>>();

export function useMyContext() {
  const value = useContext(MyContext);
  if (!value) {
    throw new Error("MyContext not found");
  }
  return value;
}

export default function Layout() {
  const [value, setValue] = createSignal("initial");

  onMount(() => {
    setValue("mutate");
  });

  return (
    <MyContext.Provider value={value}>
      <div>Layout: {value()}</div>
      <Outlet />
    </MyContext.Provider>
  );
}
```

```tsx
// src/routes/a/index.tsx
import { useMyContext } from "../a";

export default function Page() {
  const value = useMyContext();
  return <div>Page: {value()}</div>;
}
```

**Comment by edivados** [comment]
Fixed in `solid-js:1.7.8` https://github.com/solidjs/solid/issues/1795

Your second example with `1.7.8`
[![Open in StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/edit/solidjs-solid-start-xehdmz?file=src%2Froutes%2Fa.tsx,src%2Froutes%2Fa%2Findex.tsx)

Your second example with `1.7.7`
[![Open in StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/edit/solidjs-solid-start-tquw9e?file=src%2Froutes%2Fa.tsx,src%2Froutes%2Fa%2Findex.tsx)

**Comment by yume-chan** [comment]
Yes it's fixed now.
